src/apps/compound/positions.ts

```diff
@@ -0,0 +1,173 @@
+import { Address } from 'viem'
+import { getClient } from '../../runtime/client'
+import { getTokenId } from '../../runtime/getTokenId'
+import { NetworkId } from '../../types/networkId'
+import { toDecimalNumber } from '../../types/numbers'
+import {
+  ContractPositionDefinition,
+  PositionsHook,
+  TokenDefinition,
+  UnknownAppTokenError,
+} from '../../types/positions'
+import {
+  compoundMulticallAbi,
+  compoundMulticallBytecode,
+} from './abis/compound-multicall'
+
+// Data from https://docs.compound.finance/
+const MARKETS: { networkId: NetworkId; address: Address }[] = [
+  {
+    // USDC
+    networkId: NetworkId['op-mainnet'],
+    address: '0x2e44e174f7D53F0212823acC11C01A11d58c5bCB',
+  },
+  {
+    // USDT
+    networkId: NetworkId['op-mainnet'],
+    address: '0x995E394b8B2437aC8Ce61Ee0bC610D617962B214',
+  },
+  {
+    // USDC
+    networkId: NetworkId['ethereum-mainnet'],
+    address: '0xc3d688B66703497DAA19211EEdff47f25384cdc3',
+  },
+  {
+    // WETH
+    networkId: NetworkId['ethereum-mainnet'],
+    address: '0xA17581A9E3356d9A858b789D68B4d866e593aE94',
+  },
+  {
+    // USDC
+    networkId: NetworkId['arbitrum-one'],
+    address: '0x9c4ec768c28520B50860ea7a15bd7213a9fF58bf',
+  },
+  {
+    // WETH
+    networkId: NetworkId['arbitrum-one'],
+    address: '0x6f7D514bbD4aFf3BcD1140B7344b32f063dEe486',
+  },
+  {
+    // USDT
+    networkId: NetworkId['arbitrum-one'],
+    address: '0xd98Be00b5D27fc98112BdE293e487f8D4cA57d07',
+  },
+]
+
+const hook: PositionsHook = {
+  getInfo() {
+    return {
+      id: 'compound',
+      name: 'Compound',
+      description: 'Compound markets',
+    }
+  },
+  async getPositionDefinitions(
+    networkId: NetworkId,
+    address: Address | undefined,
+  ) {
+    const markets = MARKETS.filter((market) => market.networkId === networkId)
+    if (!markets.length || !address) {
+      return []
+    }
+
+    const client = getClient(networkId)
+
+    const results = await client.readContract({
+      code: compoundMulticallBytecode,
+      abi: compoundMulticallAbi,
+      functionName: 'getUserPositions',
+      args: [address, markets.map((m) => m.address)],
+    })
+
+    return results.flatMap(
+      ({ baseToken, baseTokenBalance, borrowBalance, collaterals }, index) => {
+        const balances: {
+          token: Address
+          market: Address
+          balance: bigint
+          type: string
+        }[] = [
+          {
+            token: baseToken,
+            market: markets[index].address,
+            balance: baseTokenBalance,
+            type: 'Supply',
+          },
```

Comment:
Maybe this one could/should be modeled using `app-token-definition`? 
This way we would have a price for `cToken`?
i.e. the supplied tokens would show up with both a token amount and $ value in the positions.

Reply:
Ohh, right, I hadn't noticed token amounts don't show up.

I don't think we can use a token definition because the contract is not an ERC20. cTokens were used on Compound v2, but on the current live version (v3) there are no cTokens. The contract doesn't have `symbol` or `name` functions (it does seem to have the others though 🤔).

Maybe we can make it so if a contract position has only one token we can show the underlying token amount too? Or maybe create a different kind of position for one-token positions, not sure.

Reply:
Oh ok, my knowledge cutoff was at Compound v2 😄 

Yes good idea, we should update Valora so it displays the token amount when there's only one. I wanted this for other contract positions.